modules/core/shared/src/test/scala/gem/DatasetLabelSpec.scala

```diff
@@ -7,13 +7,15 @@ import cats.tests.CatsSuite
 import cats.{ Eq, Show }
 import cats.kernel.laws.discipline._
 import gem.arb._
+import gem.laws.discipline._
 
 @SuppressWarnings(Array("org.wartremover.warts.ToString"))
 final class DatasetLabelSpec extends CatsSuite {
   import ArbDataset._
 
   // Laws
   checkAll("DatasetLabel", OrderTests[Dataset.Label].order)
+  checkAll("Optics.fromString", FormatTests(Dataset.Label.Optics.fromString).formatWith(strings))
```

Comment:
Is `formatWith` part of scalacheck? or did you add it to the test package?

Reply:
It's our code but it follows the [pattern for law checking](https://typelevel.org/cats/typeclasses/lawtesting.html) used by cats. `FormatTests` is a factory that constructs a `RuleSet` that's evaluated by Scala(Test|check).